pyteal/ast/subroutine_test.py

```diff
@@ -78,6 +78,76 @@ def fnWithPartialExprAnnotations(a, b: Expr) -> Expr:
         assert invocation.args == args
 
 
+def test_subroutine_invocation_param_types():
+    def fnWithNoAnnotations(a, b):
+        return Return()
+
+    def fnWithExprAnnotations(a: Expr, b: Expr) -> Expr:
+        return Return()
+
+    def fnWithSVAnnotations(a: ScratchVar, b: ScratchVar):
+        return Return()
+
+    def fnWithMixedAnns1(a: ScratchVar, b: Expr) -> Expr:
+        return Return()
+
+    def fnWithMixedAnns2(a: ScratchVar, b) -> Expr:
+        return Return()
+
+    def fnWithMixedAnns3(a: Expr, b: ScratchVar):
+        return Return()
+
+    sv = ScratchVar()
+    x = Int(42)
+    s = Bytes("hello")
+    cases = [
+        ("vanilla 1", fnWithNoAnnotations, [x, s], None),
+        ("vanilla 2", fnWithNoAnnotations, [x, x], None),
+        ("vanilla no sv's allowed 1", fnWithNoAnnotations, [x, sv], TealInputError),
+        ("exprs 1", fnWithExprAnnotations, [x, s], None),
+        ("exprs 2", fnWithExprAnnotations, [x, x], None),
+        ("exprs no sv's asllowed 1", fnWithExprAnnotations, [x, sv], TealInputError),
```

Comment:
@tzaffi Friendly pointer to minor typo.

```suggestion
        ("exprs no sv's allowed 1", fnWithExprAnnotations, [x, sv], TealInputError),
```